lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs

```diff
@@ -47,84 +43,43 @@ namespace Org.Apache.REEF.Evaluator
 {
     public sealed class Evaluator
     {
-        private static Logger logger = Logger.GetLogger(typeof(Evaluator));
-        private static int heartbeatPeriodInMs = Constants.DefaultEvaluatorHeartbeatPeriodInMs;
-        private static int heartbeatMaxRetry = Constants.DefaultEvaluatorHeartbeatMaxRetry;
-        private static IInjector injector;
-        private static EvaluatorConfigurations evaluatorConfig;
+        private static Logger _logger = Logger.GetLogger(typeof(Evaluator));
+
+        [Inject]
+        private Evaluator(
+            RuntimeClock clock,
+            EvaluatorRuntime evaluatorRuntime,
+            HeartBeatManager heartBeatManager,
+            ContextManager contextManager,
+            CustomTraceListeners customTraceListeners,
+            CustomTraceLevel customTraceLevel)
+        {
+            SetCustomTraceListeners(customTraceListeners, customTraceLevel);
+            heartBeatManager.Evaluatorruntime = evaluatorRuntime;
+            heartBeatManager.Contextmanager = contextManager;
+            SetRuntimeHandlers(evaluatorRuntime, clock);
+            clock.Run();
```

Comment:
Is this safe in C#? I'd expect a method `Run()` or such in `Evaluator` that is called in `Main`.